### ISSUE TITLE
Advanced Search Selectors - 2.5.3 Label in Name - None of the menu items in any select2 dropdown has an accessible label

### ISSUE THREAD

**Issue by imogenroberts** [archesproject/arches#10107]
As a user reliant on the screen reader to discern the options provided in a list/menu, I expect all menu items to have accessible labels given to them in the HTML which can then be interpreted by the screen reader to exactly match what is displayed visually.

**Linked Work Item:**
#9386 

**Expected Result:**
Every menu item should have an accessible label to ensure that they are read out by the screen reader exactly as they appear visually.

**Actual Result:**
All the menu items in all select2 dropdowns are read as 'Blank', so users with vision difficulties will have no idea what to pick from the menu. This issue is exacerbated by the lack of accessible labels for each dropdown menu - not only are the menu items not clear, the user can't even use the title of the menu to guess at what the menu items might be. 

**Evidence:**

![Image](https://github.com/archesproject/arches/assets/77161085/3a2cb8a0-1c4e-4a6e-9c03-2d24c54b4cbb)

**Comment by apeters** [comment]
@imogenroberts this should be fixed in the latest.

**Comment by imogenroberts** [comment]
RETEST: All menu items in the Advanced Search panel now have accessible labels which match their visual representation.

**Evidence:**

https://github.com/archesproject/arches/assets/77161085/da259aa6-b967-482e-9fa5-b8f5ea0352d0
